saleor/graphql/warehouse/dataloaders.py

```diff
@@ -206,6 +211,55 @@ def batch_load_stocks_by_country(
         ]
 
 
+class StocksReservationsByCheckoutTokenLoader(DataLoader):
+    context_key = "stock_reservations_by_checkout_token"
+
+    def batch_load(self, keys):
+        from ..checkout.dataloaders import CheckoutLinesByCheckoutTokenLoader
+
+        def with_checkouts_lines(checkouts_lines):
+            checkouts_keys_map = {}
+            for i, key in enumerate(keys):
+                for checkout_line in checkouts_lines[i]:
+                    checkouts_keys_map[checkout_line.id] = key
+
+            def with_lines_reservations(lines_reservations):
+                reservations_map = defaultdict(list)
+                for reservations in lines_reservations:
+                    for reservation in reservations:
+                        checkout_key = checkouts_keys_map[reservation.checkout_line_id]
+                        reservations_map[checkout_key].append(reservation)
+
+                return [reservations_map[key] for key in keys]
```

Comment:
Dataloader should always return a list with the same length as the length of keys. 
```suggestion
                return [reservations_map.get(key) for key in keys]
```

Reply:
`reservations_map` is a `defaultdict` that returns list for nonexistant keys (`.get(key, list())`) and `keys` here is argument passed to `batch_load` so it will always be list of reservations found for given checkout tokens (or empty lists if reservations weren't found for given checkout.